app/helpers/users_helper.rb

```diff
@@ -1,2 +1,8 @@
 module UsersHelper
+  # Returns the Gravatar for the given user.
+  def gravatar_for user, size: Settings.user.size
+    gravatar_id = Digest::MD5.hexdigest(user.email.downcase)
```

Comment:
Phân biệt cho anh downcase và downcase!

Reply:
đều là chuyển về chữ thường, nhưng mà nên dùng downcase! hơn vì khi dùng phương thức đó nó sẽ thay đổi giá trị đó về in thường luôn, khi gọi lại thì nó vẫn còn là in thường, còn downcase thì chỉ in thương khi ta gọi phương thức downcase. còn khi k gọi phương thức đó thì nó quay về giá trị như ban đầu
ví dụ : 
downcase!:
a = "AAAA"
a.downcase!
puts a
=>"aaaa"
còn downcase:
b = "BBBBB"
puts b.downcase
=>"bbbbb"
puts b
=>"BBBBB"

Reply:
thế nếu chuỗi không có kí tự in hoa thì sao em

Reply:
Câu này em reply lại nhé "thế nếu chuỗi không có kí tự in hoa thì sao em"

Reply:
nếu là 1 chuỗi k có kí tự in hoa thì nó vẫn giữ nguyên ạ